### PR TITLE
C51-417: Configurable 'Add Cases' Button on Contact 

### PR DESCRIPTION
**Pull request by tunbola**
## Overview
This PR allows to make the 'Add Case' button on a contact  summary page trigger a webform instead of the modal 'New Case' modal that is triggered by default.
A new configuration is added for this at `civicrm/admin/setting/case?reset=1` that allows a user to configure if clicking the `Add Case` button should trigger a webform or not. If yes, a text box is provided to input the webform URL.

## Before
The `Add Case` button only opens the add case modal.

## After
The `Add Case` button  can either open the add case modal or a web form URL depending on the configuration.

civicaseAddwebform

### CODE REVIEW

**Comment by shitijg** [comment]
@tunbola I have also added some labels / help text in the JIRA ticket - so if you can add those here - that would be great. 

Also, a screengrab of the add case button leading to an actual webform would be great too. 

Thanks for your work.

**Comment by jamienovick** [comment]
Does this also cover the new case functionality from the main menu and any other new case links? Really I would suggest we redirect any "new case" link to this webform. Thanks

**Comment by shitijg** [comment]
This only changes the button on the Contact Tab. The new case functionality would be a bit different as it does not pertain to a particular contact - so we will not pass the contactID in the URL. 

@jamienovick do you want to add this there? This would also mean that we can't release this today.

**Comment by jamienovick** [comment]
We will need a suitable solution for new case not from contact, where will we document it so it does not get missed?

**Comment by shitijg** [comment]
I'll create another ticket and add it to Rene's backlog for next week.

File: ang/civicase.ang.php [hunk 1]
```diff
@@ -235,6 +235,14 @@
 $options['allowMultipleCaseClients'] = (bool) Civi::settings()->get('civicaseAllowMultipleClients');
 $options['allowCaseLocks'] = (bool) Civi::settings()->get('civicaseAllowCaseLocks');
 
+// Retrieve civicase webform URL
+$allowCaseWebform = Civi::settings()->get('civicaseAllowCaseWebform');
+$webformURl = NULL;
+if ($allowCaseWebform) {
+  $webformURl = Civi::settings()->get('civicaseWebformUrl');
+}
```
Comment (reneolivo) [inline ang/civicase.ang.php]: Do you think it's possible to do this in PHP?

```php
$options['webformUrl'] = $allowCaseWebform
  ? Civi::settings()->get('civicaseWebformUrl')
  : NULL;
```

File: ang/civicase.ang.php [hunk 2]
```diff
@@ -235,6 +235,14 @@
...
+$options['webformUrl'] = $webformURl;
```
Comment (reneolivo) [inline ang/civicase.ang.php]: I think `newCaseWebformUrl` would be a bit more explicit, unless this is used for something else other than creating new cases.

File: ang/civicase.ang.php (hunk 1 above)
Comment (tunbola) [reply ang/civicase.ang.php]: Yes we can!

**Comment by tunbola** [comment]
@shitijg , As discussed, the label is added appropriately already. I ran this on a clean civi build and don't have webform and all installed but will work fine as long as URL exists.

**Pull request merged**